airflow/providers/openlineage/extractors/base.py

```diff
@@ -17,27 +17,34 @@
 
 from __future__ import annotations
 
+import warnings
 from abc import ABC, abstractmethod
-from typing import TYPE_CHECKING
+from typing import Generic, TypeVar, Union
 
 from attrs import Factory, define
+from openlineage.client.event_v2 import Dataset as OLDataset
+
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore", DeprecationWarning)
+    from openlineage.client.facet import BaseFacet as BaseFacet_V1
```

Comment:
I'm not i understand this import from v1 and the logic that follows it, why do we still need it? If somebody uses the new provider version, they must use the OL client that support v2 facets right? I must be missing something here 😄